File: python/002-LaSucesionDeFibonacci.py
''' 
* Escribe un programa que imprima los 50 primeros números de la sucesión
 * de Fibonacci empezando en 0.
 * - La serie Fibonacci se compone por una sucesión de números en
 *   la que el siguiente siempre es la suma de los dos anteriores.
 *   0, 1, 1, 2, 3, 5, 8, 13...
 '''
''' Primer intento
def Fibonacci(n):
    if n ==0:
        return None
    else:
        fibo = [0,1]
        fibo.append(1)
        for i in range(2,n):
            fibo.append(fibo[i]+fibo[i-1])
        return fibo
'''

def Fibonacci(n):
    """
    Generate the Fibonacci sequence up to the n-th term.

    Args:
        n (int): The number of terms to generate in the Fibonacci sequence.

    Returns:
        list: The Fibonacci sequence up to the n-th term.
    """
    fibo = [0, 1]  # Initialize the Fibonacci sequence with the first two terms
    for i in range(n - 2):  # Generate the remaining terms in the sequence
        fibo.append(fibo[i] + fibo[i + 1])
    return fibo

print(Fibonacci(50))
print(len(Fibonacci(50)))

File: javascript/002-LaSucesionDeFibonacci.js
/*
 * Escribe un programa que imprima los 50 primeros números de la sucesión
 * de Fibonacci empezando en 0.
 * - La serie Fibonacci se compone por una sucesión de números en
 *   la que el siguiente siempre es la suma de los dos anteriores.
 *   0, 1, 1, 2, 3, 5, 8, 13...
 */

function Fibonacci(n) {
    const fibo = [0, 1]; // inicializamos con los dos primeros términos

    for (let i = 2; i < n; i++) {     // comenzamos desde el tercero, en js empieza en 0 y acaba en n-1
        // Se puede acabar en n cambiando la condición a i <= n
        // se añade elemento a la lista al final, la suma de los dos anteriores.
        fibo.push(fibo[i-1] + fibo[i-2]);
    }
    return fibo;
}

console.log(Fibonacci(50));
console.log(Fibonacci(50).length);

